### PR TITLE
imporvement on routes response 

### PR DESCRIPTION
**Pull request by horix7**
### Fixing validation on setting the right status code 

> worked on the feedback and improved on user data validation
offices controllers and specifying the real respond status to the route and also setting up the params validation 

#### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

### CODE REVIEW

File: server/controllers/officeControlers.js [hunk 1]
```diff
@@ -30,21 +30,28 @@ class officeManagement {
...
+                "status": 400,
+                "message": "the id you specified is not valid "
+            }); 
+        }
+        office.viewOffice(parseInt(req.params.id).toString())
         .then(results => {
             if(results == "dont exist") {
                 return res.status(403).json({
                     "status": 403,
                     "error":"offices you are requesting does not exist exist "
```
Comment (Joe-Joseph) [inline server/controllers/officeControlers.js]: Remove one exist

File: server/controllers/officeControlers.js [hunk 2]
```diff
@@ -106,7 +120,14 @@ class officeManagement {
     }
 
     deleteOffice(req,res) {
-        office.deleteOffice(req.params.id)
+        
+        if(parseInt(req.params.id).toString() == "NaN") {
```
Comment (Joe-Joseph) [inline server/controllers/officeControlers.js]: Why are you converting that params into string for you to check if it's an integer. there is some other javascript method that can help you do that. you can use typeof or Number.isInterger(). checkout how they are used and choose one you can use.

File: server/middleware/authentication.js [hunk 3]
```diff
@@ -1,26 +1,32 @@
...
 
 
 const authenticate = (req, res, next) => {
+
+try {
   if(req.headers['authorization'] == undefined) {
     return res.status(403).json({
-      status: 'error',
-      error: 'not logged in',
+      status: 403,
```
Comment (Joe-Joseph) [inline server/middleware/authentication.js]: Is 403 the right status code for unauthorized?

**Review by Joe-Joseph — CHANGES_REQUESTED** [review]

File: server/controllers/officeControlers.js (hunk 2 above)
Comment (horix7) [reply server/controllers/officeControlers.js]: params are by default string thats why am converting them back into string to check if they are valid

**Pull request merged**